### PR TITLE
count_vowels 

### LINKED ISSUE #19: Count vowels
Takes a string as input and returns the count of vowels in the string.ring.
Vowels include a, e, i, o, u (case-insensitive).

-Example:
Input:   "Hello World"
Output: 3

-Explanation:
The vowels in the string are 'e', 'o', and 'o', totaling 3

### PR DESCRIPTION
**Pull request by AseelOmer**
---
name:  Aseel
about: count_vowels 
---
- #19

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/tests/test_count_vowels.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        self.assertEqual(count_vowels(large_input), 1000000)
+
+    def test_count_vowels_with_numbers_and_symbols(self):
+        """Test that count_vowels correctly counts vowels with numbers and symbols."""
+        self.assertEqual(count_vowels("1234!@#$%^&*()"), 0)
+
+    def test_count_vowels_mixed_case_and_symbols(self):
+        """Test that count_vowels correctly handles mixed case with symbols."""
+        self.assertEqual(count_vowels("Aeiou123!"), 5)
+
```
Comment (SiSaR-Pal) [inline solutions/tests/test_count_vowels.py:62]: Add a test for input with whitespace to ensure it's handled correctly (e.g., " A e i o u ").

**Review by SiSaR-Pal — DISMISSED** [review]
The code and tests are well-written, maintainable, and align with the provided template. The function is robust, and the unit tests thoroughly cover typical cases, edge cases, and invalid inputs. Excellent work!

File: solutions/tests/test_count_vowels.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import unittest
+from solutions.count_vowels import count_vowels
+
```
Comment (Nada-saad635) [inline solutions/tests/test_count_vowels.py:18]: if you can write here an docstring  explanation of the program what it does will be helpful

Comment (AseelOmer) [reply solutions/tests/test_count_vowels.py:18]: will do right now

File: solutions/tests/test_count_vowels.py (hunk 1 above)
Comment (AseelOmer) [reply solutions/tests/test_count_vowels.py:62]: Thankyou so much for the review, really appreciate it, will add that now

**Review by omniaNS — COMMENTED** [review]
Nice work! I have one small suggestion to enhance the readability of your test file: consider adding short comments to differentiate defensive tests from the standard ones. Keep up the good work! 👏

**Comment by Nada-saad635** [comment]
good work!

**Review by Nada-saad635 — APPROVED** [review]
good work Aseel !! ready to merge
